### PR TITLE
clean up status handling in `css.Job`

### PR DESCRIPTION
**Pull request by richrines1**
* consistent use of `gss.SuperstaqUnsuccessfulJobException` for failed jobs
* improve the error message in said exception
* raise TImeoutError if `job.counts()` times out but the job hasn't failed
* fix mocked statuses in css.job_test which previously weren't any of those in Job.ALL_STATES (in one case causing a test to pass which shouldn't have)

### CODE REVIEW

File: cirq-superstaq/cirq_superstaq/job.py [hunk 1]
```diff
@@ -75,9 +75,12 @@ def _refresh_job(self) -> None:
 
     def _check_if_unsuccessful(self) -> None:
         if self.status() in self.UNSUCCESSFUL_STATES:
```
Comment (dowusu-antwi) [inline cirq-superstaq/cirq_superstaq/job.py:77]:
```suggestion
        if status := self.status() in self.UNSUCCESSFUL_STATES:
```

File: cirq-superstaq/cirq_superstaq/job.py [hunk 2]
```diff
@@ -75,9 +75,12 @@ def _refresh_job(self) -> None:
...
-            raise gss.superstaq_exceptions.SuperstaQUnsuccessfulJobException(
-                self.job_id(), self.status()
-            )
+            status = self.status()
```
Comment (dowusu-antwi) [inline cirq-superstaq/cirq_superstaq/job.py:78]: Assuming the above:
```suggestion
```

**Review by dowusu-antwi — COMMENTED** [review]
Should the internals in `css.Job` start with an underscore? E.g., `TERMINAL_STATES`, `NON_TERMINAL_STATES`, etc.

**Comment by richrines1** [comment]
> Should the internals in `css.Job` start with an underscore? E.g., `TERMINAL_STATES`, `NON_TERMINAL_STATES`, etc.

imo those are meant in part to provide information/functionality to the user, so shouldn't really be considered "internals". they even [have docstrings](https://github.com/SupertechLabs/client-superstaq/blob/cfd1235c55a36161040a15040fb50e3b744ec269/cirq-superstaq/cirq_superstaq/job.py#L35-L40)

**Review by dowusu-antwi — APPROVED** [review]
> imo those are meant in part to provide information/functionality to the user, so shouldn't really be considered "internals". they even [have docstrings](https://github.com/SupertechLabs/client-superstaq/blob/cfd1235c55a36161040a15040fb50e3b744ec269/cirq-superstaq/cirq_superstaq/job.py#L35-L40)

Makes sense. LGTM.

**Pull request merged**